File: 2389-Longest-Subsequence-With-Limited-Sum/2389-Longest-Subsequence-With-Limited-Sum.cpp
class Solution {
public:
    vector<int> answerQueries(vector<int>& nums, vector<int>& queries) {
        sort(nums.begin(), nums.end());
        vector<int> result;
        for (int query : queries) {
            int count = 0;
            int total = 0;
            for (int num : nums) {
                total += num;
                count++;
                if (total > query) {
                    count--;
                    break;
                }
            }
            result.push_back(count);
        }
        
        return result;
    }
};

/* The time complexity is O(nm log n), where n is the length of the nums array and m is the length of the queries array.
This is because the solution first sorts the nums array in non-decreasing order, which takes O(n log n) time,
and then it iterates through each query and each element in the nums array, which takes O(nm) time.*/

/*The space complexity of the solution is O(m), where m is the length of the queries array.
This is because the solution creates a new vector to store the result, which takes O(m) space*/

File: 2389-Longest-Subsequence-With-Limited-Sum/2389-Longest-Subsequence-With-Limited-Sum.py
class Solution:
    def answerQueries(self, nums: List[int], queries: List[int]) -> List[int]:
        nums.sort()
        result = []
        for query in queries:
            count = 0
            total = 0
            for num in nums:
                total += num
                count += 1
                if total > query:
                    count -= 1
                    break
            result.append(count)
        return result
